apis/supernova/src/handler/hypercube-handler.js

```diff
@@ -70,10 +54,55 @@ class HyperCubeHandler extends DataPropertyHandler {
   }
 
   getDimensionLayouts() {
-    const hc = getHyperCube(this.layout, this.path);
+    const hc = hcHelper.getHyperCube(this.layout, this.path);
     return hc ? hc.qDimensionInfo : [];
   }
 
+  addDimension(dimension, alternative, idx) {
+    const dim = initializeField(dimension);
+
+    if (hcHelper.isDimensionAlternative(this, dim, alternative)) {
+      return hcHelper.addAlternativeDimension(this, dim, idx);
+    }
+
+    return addMainDimension(this, dim, idx);
+  }
+
+  async addDimensions(dimensions, alternative = false) {
+    const existingDimensions = this.getDimensions();
+    const addedDimensions = [];
+    let addedActive = 0;
+
+    await dimensions.reduce(async (prevPromise, dimension) => {
```

Comment:
While using `reduce` in this way will work. I do think you should use `for await` loop instead as it's easier to read and built to do what you want.

Reply:
I seem like a hack to me. If someone in the future where to look at this they might wonder why reduce is being used.

IMO it's fine to just ignore the lint error. The error is not valid in this case.